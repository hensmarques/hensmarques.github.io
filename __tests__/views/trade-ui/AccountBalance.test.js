import React from 'react';
import { AccountBalance } from '../../../src/views/trade-ui/AccountBalance';

describe('<AccountBalance/>', () => {
  const selectedInstrument = {
    base: 'BTC',
    quote: 'USD'
  };
  const balances = {
    USD: {total: 5000, trading: 4000},
    BTC: {total: 2, trading: 2}
  };
  const products = {
    USD: {decimals: 2},
    BTC: {decimals: 2}
  };
  const loadData = jest.fn();
  
  it('renders without crashing and fires loadData on mount', () => {
    shallow(<AccountBalance loadData={loadData} selectedInstrument={selectedInstrument} balances={balances} products={products} />);
    expect(loadData).toHaveBeenCalled();
  });

  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<AccountBalance loadData={loadData} selectedInstrument={selectedInstrument} balances={balances} products={products} />);
    expect(wrapper).toMatchSnapshot();
  });
});