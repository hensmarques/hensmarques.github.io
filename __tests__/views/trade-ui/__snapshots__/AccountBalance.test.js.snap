// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountBalance/> renders and matches the snapshot 1`] = `
<Fragment>
  <div
    className="account-balance-header"
  >
    <span
      className="account-balance-product"
    >
      Product
    </span>
    <span
      className="account-balance-available"
    >
      Available
    </span>
    <span
      className="account-balance-total"
    >
      Total
    </span>
  </div>
  <div
    className="account-balance-rows"
  >
    <div
      className="account-balance-row quote-currency-stats"
    >
      <span
        className="account-balance-product"
      >
        USD
      </span>
      <span
        className="account-balance-available"
      >
        4000
      </span>
      <span
        className="account-balance-total"
      >
        5000
      </span>
    </div>
    <div
      className="account-balance-row base-currency-stats"
    >
      <span
        className="account-balance-product"
      >
        BTC
      </span>
      <span
        className="account-balance-available"
      >
        2
      </span>
      <span
        className="account-balance-total"
      >
        2
      </span>
    </div>
  </div>
  <div
    className="account-balance-header all-balances-header"
  >
    <span
      className="account-balance-product"
    />
  </div>
  <div
    className="all-balances-container"
  >
    <ScrollBar
      className=""
      component="div"
      containerRef={[Function]}
    />
  </div>
</Fragment>
`;
